### PR TITLE
WIP: A rough start at a visualizer

### PR DESCRIPTION
**Pull request by codersasha**
A roguh start at a visualizer in graph.js, based on a template for go.js and only rendering the first 500 wires

### CODE REVIEW

**Comment by mithro** [comment]
Could you add a screenshot?

**Comment by mithro** [comment]
Can you use the [prj-xray-db](https://github.com/SymbiFlow/prjxray-db) for those files?

**Comment by mcmasterg** [comment]
Isn't prj-x-ray-db only for data files? I thought we didn't want people
committing to that?

**Comment by mithro** [comment]
@mcmasterg I believe those files are copied /from/ the DB?

**Comment by mcmasterg** [comment]
I might have misunderstood your comment, nvm

File: prjxray-wire-viewer/graph.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<html>
+<head>
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<title>FPGA visualization</title>
+<meta name="description" content="FPGA visualization." />
+<meta charset="UTF-8">
+</head>
+<body>
+<div id="myDiagramDiv" style="width:100%; height:800px; border:1px solid black"></div>
+<script src="go.js"></script>
```
Comment (mithro) [inline prjxray-wire-viewer/graph.html:12]: go.js should be in `third_party` somewhere...

File: prjxray-wire-viewer/graph.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<script src="tileconn.js"></script>
+<script src="tilegrid.js"></script>
```
Comment (mithro) [inline prjxray-wire-viewer/graph.html:14]: The `tileconn.js` and `tilegrid.js` come from the [`prjxray-db`](https://github.com/SymbiFlow/prjxray-db) right? Could you include them via adding a git submodule for prjxray-db in third_party too?

File: prjxray-wire-viewer/graph.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
```
Comment (mithro) [inline prjxray-wire-viewer/graph.html:1]: nit: Extra new line :-P

File: prjxray-wire-viewer/script.js [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+                type_tile.right.leftArray.push({portColor: 'black', portId: type_tile.name + '-' + wireDest});
+              }
+            }
+          }
+        }
+      }
+    }
+
+    console.log("Graph built. Rendering...");
+
```
Comment (mithro) [inline prjxray-wire-viewer/script.js:551]: nit: Lots of extra spaces :-P

**Review by mithro — COMMENTED** [review]
Awesome work @sbermeister -- looking forward to seeing this working 100%!

**Comment by mithro** [comment]
From the following page it looks like [`go.js` is not open source](https://gojs.net/latest/doc/download.html). This means we can't really use it - sorry :-/

**Pull request closed without merging**